schema-extended/arxiv-extended.json

```diff
@@ -0,0 +1,623 @@
+          "msc_classes": {
+            "title": "List of MSC classes",
+            "description": "Classifications from American Mathematical Society Mathematical Subject Classification (MSC).",
+            "type": "string",
+            "minLength": 1,
+            "examples": [
+              "Primary 14N35, Secondary 14H10, 14H20, 14J26",
+              "Primary 46F30. Secondary: 44A10, 46S10, 46F10, 12J15, 12J25, 16W60",
+              "46B20, 47H10, 54B20, 54F15, 68U05"
+            ],
+            "$comment": "TODO: consider array of an enumerated definition instead. Legacy input is inconsistent."
+
+          },
+          "acm_classes": {
+            "title": "List of ACM classes",
+            "description": "Classifications from ACM Computing Classification System.",
+            "type": "string",
+            "minLength": 1,
+            "examples": [
+              "C.2.0; C.2.2; C.2.3; C.2.6",
+              "I.3.5",
+              "G.1.6; F.1.1; I.2.6; G.1.3"
+            ],
+            "$comment": "TODO: consider array of an enumerated definition instead. Legacy input is inconsistent."
+          },
+          "author_links": {
+            "title": "List of author links",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/author_link"
+            },
+            "minItems": 1,
+            "uniqueItems": true,
+            "$comment": "should this be version-specific?"
+          },
+          "external_publication_status": {
+            "$ref": "#/definitions/external_publication_status"
+          },
+          "source_info": {
+            "title": "Source information",
+            "description": "Information about the input source files.",
+            "$ref": "#/definitions/source_info"
+          }
+        },
+        "required": [
+          "identifier",
+          "submitted_date",
+          "publication_date",
+          "license",
+          "content_types",
+          "titles",
+          "abstracts",
+          "legacy_authors",
+          "authors",
+          "arxiv_status",
+          "source_info",
+          "submitter"
+        ]
+      }
+    },
+
+    "required": [
+      "identifier",
+      "primary_classification",
+      "versions"
+    ],
+    "properties": {
+      "identifier": {
+        "title": "arXiv identifier",
+        "description": "Canonical arXiv e-print identifier, without a version affix.",
+        "$ref": "#/definitions/unversioned_identifier"
+      },
+      "primary_classification": {
+        "title": "Primary classification",
+        "description": "Primary classification of this document.",
+        "$ref": "#/definitions/category",
+        "examples": [
+          "cs.AI",
+          "hep-th"
+        ]
+      },
+      "secondary_classifications": {
+        "title": "Secondary classifications",
+        "description": "Optional and unranked secondary classifications of this document. Must be sorted alphabetically.",
```

Comment:
Maybe we can drop the alph sorting requirement for NG? It would be nice if the only validation that was needed was in the schema.

Reply:
I don't think we'll be able to completely avoid custom validation.